File: v2/harvest_api_sample.php
<?php
  $url = "https://api.harvestapp.com/v2/users/me";
  $headers = array(
    "Authorization: Bearer " . getenv("HARVEST_ACCESS_TOKEN"),
    "Harvest-Account-ID: "   . getenv("HARVEST_ACCOUNT_ID")
  );

  $handle = curl_init();
  curl_setopt($handle, CURLOPT_URL, $url);
  curl_setopt($handle, CURLOPT_RETURNTRANSFER, 1);
  curl_setopt($handle, CURLOPT_HTTPHEADER, $headers);
  curl_setopt($handle, CURLOPT_USERAGENT, "PHP Harvest API Sample");

  $response = curl_exec($handle);

  if (curl_errno($handle)) {
    print "Error: " . curl_error($handle);
  } else {
    print json_encode(json_decode($response), JSON_PRETTY_PRINT);
    curl_close($handle);
  }
?>

File: v2/harvest_api_sample.py
import os
import json
import urllib.request

url = "https://api.harvestapp.com/v2/users/me"
headers = {
    "User-Agent": "Python Harvest API Sample",
    "Authorization": "Bearer " + os.environ.get("HARVEST_ACCESS_TOKEN"),
    "Harvest-Account-ID": os.environ.get("HARVEST_ACCOUNT_ID")
}

request = urllib.request.Request(url=url, headers=headers)
response = urllib.request.urlopen(request, timeout=5)
responseBody = response.read().decode("utf-8")
jsonResponse = json.loads(responseBody)

print(json.dumps(jsonResponse, sort_keys=True, indent=4))
